### PR TITLE
feat(cmf): add saga to handle http actions

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

We have redux middleware to handle http action which is doing fetch
We have http saga which are doing fetch.

code duplication with some tiny difference:

middleware handle onError by default and propose to override it
middleware handle onResponse + transform options

**What is the chosen solution to this problem?**

Need #1293

To migrate that we will first add a saga to handle http actions
We will let the middleware to do some warning if you are in some corner case

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/src/sagas/http.js [hunk 1]
```diff
@@ -286,6 +287,78 @@ export function getDefaultConfig() {
...
+ * saga forked at bootstrap to take all actions.http signal
+ */
+function* takeAllHTTPAction() {
+	yield takeEvery(HTTP_METHODS.DELETE, handleHTTPAction);
+	yield takeEvery(HTTP_METHODS.GET, handleHTTPAction);
+	yield takeEvery(HTTP_METHODS.POST, handleHTTPAction);
+	yield takeEvery(HTTP_METHODS.PUT, handleHTTPAction);
+	yield takeEvery(HTTP_METHODS.PATCH, handleHTTPAction);
+	yield take('DO_NOT_QUIT');
+}
```
Comment (ncomont) [inline packages/cmf/src/sagas/http.js]: Could be replaced by : 

```
function* takeAllHTTPAction() {
	yield takeEvery(
		[
			HTTP_METHODS.DELETE,
			HTTP_METHODS.GET,
			HTTP_METHODS.POST,
			HTTP_METHODS.PUT,
			HTTP_METHODS.PATCH
		],
		handleHTTPAction
	);
	yield take('DO_NOT_QUIT');
}
```

https://github.com/redux-saga/redux-saga/blob/f85674b3a319c1f9c209e62bc509a0e2c0b4bdeb/packages/core/src/internal/proc.js#L448

https://github.com/redux-saga/redux-saga/blob/d98344e4fd6a7be561bd26f58d5c831e3b14c1b8/packages/core/src/internal/matcher.js#L14



... Even if I'm not sure to understand the goal of the last `take` in this case.

File: packages/cmf/src/sagas/http.js [hunk 2]
```diff
@@ -286,6 +287,78 @@ export function getDefaultConfig() {
...
+		response = yield call(httpDelete, url, config, options);
+	} else if (type === HTTP_METHODS.GET) {
+		response = yield call(httpGet, url, config, options);
+	} else if (type === HTTP_METHODS.POST) {
+		response = yield call(httpPost, url, body, config, options);
+	} else if (type === HTTP_METHODS.PUT) {
+		response = yield call(httpPut, url, body, config, options);
+	} else if (type === HTTP_METHODS.PATCH) {
+		response = yield call(httpPatch, url, body, config, options);
+	}
```
Comment (ncomont) [inline packages/cmf/src/sagas/http.js]: Maybe this could be achieved by looping on every HTTP methods types ?

File: packages/cmf/src/sagas/http.js [hunk 3]
```diff
@@ -286,6 +287,78 @@ export function getDefaultConfig() {
...
+	if (!response.response.ok) {
```
Comment (ncomont) [inline packages/cmf/src/sagas/http.js]: I'd like to talk with you about what should be an HTTP error for the HTTP middleware. I'm not sure to agree with the actual implementation, and I'd like to understand why this has been done like that.

File: packages/cmf/src/sagas/http.js [hunk 4]
```diff
@@ -286,6 +287,78 @@ export function getDefaultConfig() {
...
+			yield put(onResponse(data));
+		} else if (typeof onResponse === 'string') {
+			yield put({
+				type: onResponse,
+				response: response.data,
+			});
+		}
+		if (cmf) {
+			if (cmf.collectionId) {
+				yield put(actions.collections.addOrReplace(cmf.collectionId, data));
```
Comment (ncomont) [inline packages/cmf/src/sagas/http.js]: Too deeply imbricated in my opinion.
